CodingTracker.DreamFXX/DatabaseManager.cs

```diff
@@ -0,0 +1,138 @@
+                                        Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                        Date TEXT,
+                                        StartTime TEXT,
+                                        EndTime TEXT,
+                                        Duration TEXT)");
+            }
+        }
+
+        public void InsertRecord(string? date, string? startTime, string? endTime, string duration)
+        {
+            if (string.IsNullOrWhiteSpace(date) || string.IsNullOrWhiteSpace(startTime) || string.IsNullOrWhiteSpace(endTime))
+            {
+                throw new ArgumentException("Error - You must provide a date, start time, and end time in displayed format.");
+            }
+
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+                connection.Execute(
+                    @"INSERT INTO MyCodingTracker (Date, StartTime, EndTime, Duration)
+                          VALUES (@date, @startTime, @endTime, @duration)",
+                    new {date, startTime, endTime, duration });
+            }
+        }
+
+        public void DeleteRecord(string? date)
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+
+                connection.Execute(@"DELETE FROM MyCodingTracker
+                                            WHERE Date = @date", new { date });
+            }
+        }
+
+        public void UpdateRecord(string? oldDate, string? newDate, string? startTime, string? endTime, string? duration)
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+
+                if (@startTime is null && @endTime is null)
+                {
+                    connection.Execute(@"UPDATE MyCodingTracker
+                                             SET Date = @newDate
+                                             WHERE Date = @oldDate", new {newDate,oldDate});
+                }
+                else if (@newDate is null)
+                {
+                    connection.Execute(@"UPDATE MyCodingTracker
+                                             SET StartTime = @startTime, EndTime = @endTime, Duration = @duration
+                                             WHERE Date = @oldDate", new {startTime, endTime, duration, oldDate});
+                }
+                else
+                {
+                    connection.Execute(@"UPDATE MyCodingTracker
+                                              SET Date = @newDate, StartTime = @startTime, EndTime = @endTime, Duration = @duration
+                                              WHERE Date = @oldDate", new {newDate, startTime, endTime, duration, oldDate});
+                }
+            }
+        }
+
+        public List<CodingSession> ReadFromDb()
```

Comment:
🔴 **This is still not resolved**

Note:
Dapper vs ADO.NET

![image](https://github.com/user-attachments/assets/2f090f03-f242-4121-b6e2-c7d5c85bb29e)

Reply:
I dont get it, i dont have this on my local storage.

Reply:
Ill look onto it

Reply:
@DreamFXX, 

If you are saying that my screenshots, don't match your local code. Well, that because I wrote the changes in code of what I was expecting your update to be. And then screenshotted it.

If you are saying that you already made this change and the PR doesn't reflect your local copy then attempt a commit and push.

Otherwise, just upload a screenshot of your local code, of the two methods, as a comment in this PR. And I will manually review and sign off.

Thanks and good luck 👍

Reply:
already submitted the final fixed version. ill upload a screenshot. thanks @chrisjamiecarter 
![ViewAllMethod (1)](https://github.com/user-attachments/assets/fabe0f0b-4a70-460c-bf99-f969eed76e7b)
![ViewOneMethod (2)](https://github.com/user-attachments/assets/780d49a4-236d-4b7d-adb5-8dd578b05df3)

Reply:
![DbManager](https://github.com/user-attachments/assets/3f0d9e87-4433-4c7b-a8bd-13095eeebd88)